shared/ssl.sh

```diff
@@ -64,9 +72,31 @@ CERTSUBJECT="/CN=${CERTNAME}"
 CERTHEADER="-----BEGIN CERTIFICATE-----"
 CURLFLAGS="--silent --show-error --cacert ${CACERTFILE} --retry 5 --retry-connrefused --retry-delay 2"
 
+### Handle certificate extensions
+# NOTE If we want to expand support for more extensions, it would be better
+# to define them in a .conf file rather than directly on the CLI.
+# That would also work on older versions of openssl that don't support
+# the `-addext` flag.
```

Comment:
This `-addext` flag is a convenience that is not available everywhere. If we need to, we can rewrite this to write to a .conf file instead, which is also how we should handle arbitrary extensions if we ever need to.

I verified this is available on:
- `ubuntu:16.04` with `OpenSSL 1.1.1a  20 Nov 2018` (most of our images)
- `ubuntu:18.04` with `OpenSSL 1.1.1  11 Sep 2018` (some PE images)
- `openjdk:8-jre-alpine` with `OpenSSL 1.1.1b  26 Feb 2019` (puppetdb image)
- `alpine:3.9` with `OpenSSL 1.1.1b  26 Feb 2019` (r10k image)

Unfortunately it's **not** available on:
- `alpine:3.8` with `OpenSSL 1.0.2r  26 Feb 2019` (puppet-agent-alpine image)
- my Mac OS which is actually `LibreSSL 2.6.5` ?

Reply:
And we know we can't bump `puppet-agent-alpine` to `alpine:3.9` because of the compiler shenanigans (and the fact that our JSON C++ lib won't compile on GCC 8)

Reply:
Do you think we'll move to `ssl.sh` on agent containers? I thought that was mostly to get around agent deps to generate certs

Reply:
I don't imagine we'll need to put `ssl.sh` on agent containers. Or the r10k container.

I just mentioned those images for completeness.